src/main/java/pl/psnc/dei/ui/components/batches/DatasetImportComponent.java

```diff
@@ -0,0 +1,307 @@
+        excludeSelectedCheckbox = new Checkbox();
+        excludeSelectedCheckbox.setLabel("Exclude following records");
+        excludeSelectedCheckbox.addValueChangeListener(event -> {
+            doExcludeSelected = event.getValue();
+            excludeSelectedTextArea.setEnabled(doExcludeSelected);
+        });
+        return excludeSelectedCheckbox;
+    }
+
+    private HorizontalLayout prepareAggregatorProjectDatasetSelector() {
+        HorizontalLayout horizontalLayout = new HorizontalLayout();
+        horizontalLayout.add(createAggregatorSelection());
+        horizontalLayout.add(createProjectSelection());
+        horizontalLayout.add(createDatasetsSelection());
+        return horizontalLayout;
+    }
+
+    private Component prepareDatasetNameInputField() {
+        HorizontalLayout horizontalLayout = new HorizontalLayout();
+        europeanaDatasetIdTextField = new TextField();
+        europeanaDatasetIdTextField.addClassName("flex-1");
+        europeanaDatasetIdTextField.setLabel("Europeana Dataset ID");
+        Button button = new Button();
+        button.setText("Check dataset");
+        button.getStyle().set("align-self", "flex-end");
+        button.addClickListener(event -> checkDataset());
+        horizontalLayout.add(europeanaDatasetIdTextField, button);
+        return horizontalLayout;
+    }
+
+    private Component prepareRecordLimitInput() {
+        CheckboxWithIntegerInput checkbox = new CheckboxWithIntegerInput("Limit number of records to retrieve to");
+        limitRecordsToRetrieveCheckbox = checkbox.getCheckbox();
+        limitRecordsToRetrieveIntegerField = checkbox.getIntegerField();
+        limitRecordsToRetrieveIntegerField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        limitRecordsToRetrieveCheckbox.addValueChangeListener(event -> {
+            doLimitNumberOfRecordsToRetrieve = event.getValue();
+            limitRecordsToRetrieveIntegerField.setEnabled(doLimitNumberOfRecordsToRetrieve);
+        });
+        return checkbox;
+    }
+
+    private Component prepareRecordsSplittingInput() {
+        splitRecordsCheckboxWithNumber = new CheckboxWithIntegerInput("Split records into several imports with a maximum size of");
+        splitRecordsCheckboxWithNumber.setEnabled(doCreateImport);
+        splitRecordsCheckbox = splitRecordsCheckboxWithNumber.getCheckbox();
+        splitRecordsIntegerField = splitRecordsCheckboxWithNumber.getIntegerField();
+        splitRecordsIntegerField.setEnabled(doSplitRecords);
+        splitRecordsCheckbox.addValueChangeListener(event -> {
+            doSplitRecords = event.getValue();
+            splitRecordsIntegerField.setEnabled(doSplitRecords);
+        });
+        return splitRecordsCheckboxWithNumber;
+    }
+
+    private TextField prepareImportNameInputField() {
+        importNameTextField = new TextField();
+        importNameTextField.setLabel("Import name");
+        importNameTextField.getStyle().set("padding-top", "6px");
```

Comment:
Please move such CSS settings to styles file. This remark should be applied to any other similar situation in code.

Reply:
moved styles to css file